File: hello-spark/src/main/java/com/yicj/spark/hello/HelloWorldApp.java
package com.yicj.spark.hello;

import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.function.FilterFunction;
import org.apache.spark.api.java.function.MapFunction;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;
import static org.apache.spark.sql.functions.concat;
import static org.apache.spark.sql.functions.lit;

/**
 * @author: yicj
 * @date: 2023/5/16 21:31
 */
public class HelloWorldApp {

    //https://github.com/jgperrin/net.jgp.books.spark.ch01
    public static void main(String[] args) {
        String path = "data/README.md" ;
        SparkSession spark = SparkSession.builder()
                .appName("readme app")
                .master("local[1]")
                .getOrCreate();

        Dataset<String> df = spark.read()
                .textFile(path);
        long count = df.count();
        //
        String first = df.first();
        //
        df.filter((FilterFunction<String>)line -> line.contains("Spark")) ;
        //
        long countSpark = df.filter((FilterFunction<String>) line -> line.contains("Spark"))
                .count();
        //
//        df.map(new MapFunction<String, Integer>() {
//            @Override
//            public Integer call(String value) throws Exception {
//                return null;
//            }
//        });
    }
}

File: hello-spark/src/main/scala/com/yicj/spark/scala/HelloWorldApp.scala
package com.yicj.spark.scala

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions.col

object HelloWorldApp {

  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder()
      .appName("Spark Sql basic example")
      .master("local")
      .config("spark.some.config.option", "some-value")
      .getOrCreate()
    var df = spark.read.json("data/people.json")
//    df.show() ;
//
//    df.printSchema()
//
//    df.select("name").show() ;
    //
    // df.filter(col("age") > 21).show()
    //df.groupBy("age").count().show()

   // df.createTempView("people")

//    var sqlDF = spark.sql("select * from people")
//    sqlDF.show()

    import spark.implicits._//这里spark出现了爆红

    df.select($"name", $"age" +1).show()

//    df.createGlobalTempView("people")
//    spark.sql("select * from global_temp.people").show()
//    //
//    spark.newSession().sql("select * from global_temp.people").show()

    spark.close()
  }

}
